src/theory/bv/proof_checker.cpp

```diff
@@ -63,26 +56,26 @@ Node BVProofRuleChecker::checkInternal(ProofRule id,
     Assert(args[0].getKind() == Kind::BITVECTOR_EAGER_ATOM);
     return args[0].eqNode(args[0][0]);
   }
-
+  else if (id == ProofRule::THEORY_REWRITE)
+  {
+    Assert(children.empty());
+    Assert(args.size() == 1);
+    auto const& node = args[0];
```

Comment:
This is not the proper implementation for this checker.

It should first case split on the TheoryId. Then, for the TheoryBV rules, it should return the equality if the rule applies, or otherwise fallthrough / return null.

Reply:
I see what you mean. I'll make this change.

Reply:
How would it know what is the current tid? Things like `SolveEq` will fail if the current tid is not BV, and this information is not given to `BVProofRuleChecker::checkInternal`.

Reply:
I think this will be clear once you add documentation for the proof rule.  You should add enough information to args to check the rule.  I believe this would be the TheoryRewriteId and the term to rewrite.